urlResolvers/github.js

```diff
@@ -0,0 +1,42 @@
+const URLResolver = require('./urlResolver');
+const GitHubAPIHandler = require('../sourceHandlers/github');
+
+const stripHttp = /^(http(s)?:\/\/)?(www.)?/;
+const urlRegex = /^github\.com\/([\w,-]+)\/([\w,-]+)/;
+
+const apiBaseUrl = 'https://api.github.com/';
+
+/**
+ * Creates the supported source handlers for a given URL.
+ * @class
+ * @memberof urlResolvers
+ * @augments urlResolvers.URLResolver
+ */
+class GitHubResolver extends URLResolver {
+  static getSourceHandlers(url) {
+    const sourceHandlers = [];
+    const repoInfo = this._parseURL(url);
+    if (repoInfo != null) {
+      const repoPath = `${repoInfo.owner}/${repoInfo.repoName}`;
+
+      sourceHandlers.push(new GitHubAPIHandler(apiBaseUrl, repoPath));
+    }
+
+    return sourceHandlers;
+  }
+
+  static _parseURL(url) {
```

Comment:
I don't have any problems with your naming. I just think it makes more sense to put the private functions first, so people new to the code who are reading through trying to understand the code don't have to wonder how a function works/what it does, scroll down to the function, then scroll back up to the implementing function they were originally reading.

What's your reasoning behind the private-last approach?